### PR TITLE
Facilitate output of Average Loss Convergence Table (ALCT)

### PR DESCRIPTION
**Pull request by hchagani-oasislmf**
### Facilitate output of Average Loss Convergence Table (ALCT)
The Average Loss Convergence Table (ALCT) can be generated and confidence levels can be set in the `analysis_settings.json`:
```
    "gul_summaries": [
        {
            "id": 1,
            "ord_output": {
                "alt_period": true,
                "alct_convergence": true,
                "alct_confidence": 0.95
            }
        }
    ]
```
where `alct_convergence` produces the ALCT when set to `true` and `alct_confidence` sets the confidence level for the confidence intervals (default 0.95). Average Loss Table (ALT) output is required to produce the ALCT.

### CODE REVIEW

File: oasislmf/schema/analysis_settings.json [hunk 1]
```diff
@@ -166,6 +166,18 @@
...
                                 "default": false
                             },
+			    "alct_convergence": {
+				"type": "boolean",
+				"title": "ALCT",
+				"description": "Average Loss Convergence Table (ORD Output flag), requires alt_period set to true",
+				"default": false
+			    },
+			    "alct_confidence": {
+				"type": "float",
```
Comment (sambles) [inline oasislmf/schema/analysis_settings.json:176]: the type `float` needs switching to `number`, just a quirk of OpenAPI https://swagger.io/docs/specification/data-models/data-types/#numbers

Its also worth adding min and max values to limit the valid range like: 
```
   "minimum": 0.0,
   "maximum": 1.0 
```

**Review by sambles — COMMENTED** [review]
Error when running with parquet output enabled:


Piwind: develop + [edited_analysis_settings_ord.zip](https://github.com/OasisLMF/OasisLMF/files/8871820/edited_analysis_settings_ord.zip)
```
RUNNING: oasislmf.execution.runner.run
/home/sam/repos/models/piwind/runs/losses-20220609143322/run_ktools.sh: line 440: 108041 Aborted                 (core dumped) ( aalcalc -Kil_S1_summary_palt -c output/il_S1_alct.parquet 0.85 -p output/il_S1_palt.parquet ) 2>> $LOG_DIR/stderror.err
/home/sam/repos/models/piwind/runs/losses-20220609143322/run_ktools.sh: line 30: 106870 Killed                  ktools_monitor.sh $$ $LOG_DIR

KTOOLS_STDERR:
terminate called after throwing an instance of 'parquet::ParquetException'
  what():  Column physical type mismatch.  Column 'SampleSize' has physical type 'INT32' not 'FLOAT'


STDOUT:
[OK] eve
[OK] getmodel
[OK] gulcalc
[OK] summarycalc
[OK] eltcalc
[ERROR] aalcalc - 1 processes lost
[OK] ordleccalc
Ktools Run Error - exitcode=1
 50%|████████████████████████████████████████▌                                        | 1/2 [00:05<00:05,  5.98s/it]
Ktools run Error: non-zero exit code or output detected on STDERR
Logs stored in: /home/sam/repos/models/piwind/runs/losses-20220609143322/log
```

**Comment by hchagani-oasislmf** [comment]
Thanks @sambles. `SampleSize` was miscast as a `float` in `aalcalc.cpp`. This has been fixed in the new ktools release candidate. The schema has been changed according to your suggestions. Would you mind taking another look please?

**Review by sambles — APPROVED** [review]
Thanks for the fix, looks good

File: oasislmf/schema/model_settings.json [hunk 2]
```diff
@@ -334,6 +335,7 @@
                      "plt_quantile",
                      "plt_moment",
                      "alt_period",
+		     "alt_convergence",
```
Comment (hchagani-oasislmf) [inline oasislmf/schema/model_settings.json:338]:
```suggestion
		     "alct_convergence",
```

**Pull request merged**